Improve k8s-cilium-exec.sh

simplyatul opened the pull request:
Implement few improvements to k8s-cilium-exec.sh
    
Details:
- Added a usage summary comment at the top of the script
- Introduced optional -e command-line option for explicit selection of cilium pod
- Added error handling for empty pod lists

Sample usage

```bash
./k8s-cilium-exec.sh -e cilium status --brief
Available Cilium pods:
1. cilium-9z2jh   cc0-worker
2. cilium-gkl86   cc0-control-plane
3. cilium-h22lp   cc0-worker3
4. cilium-k8hx7   cc0-worker2

Do you want to execute the command on a specific pod or (a) all pods?
Enter your choice (pod number or a): 1

==== Detail from pod cilium-9z2jh on node cc0-worker ====
OK

```

qmonnet commented on contrib/k8s/k8s-cilium-exec.sh:
```diff
@@ -18,19 +33,95 @@ function cleanup {
+                echo -e "$title\n$msg\n"
+            )&
+        done <<< "$pods_list"
+        wait
```
This part would need to be consolidated with the identical code at the end of the file - move it to a function and call that function?

simplyatul replied:
Thank you @qmonnet for pointing out this. Somehow I missed it. My bad :-(
I have pushed the changes and squashed previous commits.

qmonnet reviewed (commented):
Looks nice, but I wonder whether this script is the best place to add something like this. If we start adding specific pod selection, people will want to extend it to add more filters - thanks Marcelo for illustrating my point :slightly_smiling_face: - and I'm not sure bash is the best language to extend and maintain something like this. It's nearly something that we could want in a CLI tool, although I'm not sure the Cilium CLI should expose running direct commands from Cilium pods. Anyway, having this as a separate go utility would allow for easier filtering, or adding things like fuzzy finding when filtering through the list of pods instead of typing the pod index only, etc.

simplyatul committed:
Improve k8s-cilium-exec.sh

Details:
- Added a usage summary comment at the top of the script
- Introduced optional -e command-line option for explicit selection of cilium pod
- Added error handling for empty pod lists

Signed-off-by: Atul Thosar <atulthosar@gmail.com>

qmonnet reviewed (approved):
I'm still uncertain about this script being the right place for this, but André approved and I trust his judgement, and we can always move it to something else later if necessary. OK for me - Thanks for addressing my comment, and thanks a lot for the PR!
